### PR TITLE
#756 create functional listing page

### PR DESCRIPTION
**Pull request by Tokle89**
## Title
[Create functional listing page#756](https://github.com/NoroffFEU/agency.noroff.dev/issues/756)

## Describe your changes: 
fixed ad error in the routher function, that caused to trigger the funcitons for the "listings" page and not the "listing page"
created a render function that fetches the listing using querystring t o get the id. and then render and display said listing
have created and if statement that displays de "apply" btn if a student watches the listing, and removes the btn if a company watches the listing.
if the user is the author a "edit" and "delete" btn is shown

## Checklist before requesting a review

* [x] I created a new branch before starting with this ticket?
* [x] I commented the issue after finish in the Frontend Team Board?
* [x] I moved the card to Quality Assurance after finish?

## Did you run into any issues?

- Yes, I encountered issues. (Please describe below).
Did not get the class Store function to local storage to wprk, so i used the global localstorage function to fetch the user info from local storage

## Screenshots:

- No screenshots to include.

## Additional information:
The apply for job, edit, and delete btn has no functionality, as this should be created for another ticket.
the site my require some smaller scss styling

## Feedback

- Yes, I want feedback.

### CODE REVIEW

**Comment by Tokle89** [comment]
I have fixed the link for the <a> tags on the listing cards, on the listings page.
from:  `    'a',
    ['bg-theme-primary', 'text-theme-black', 'px-3', 'text-decoration-none'],
    null,
    'View',
    '../../..//pages/listings/listing/index.html'
  );`
  
  to: `const a = createElement(
    'a',
    ['bg-theme-primary', 'text-theme-black', 'px-3', 'text-decoration-none'],
    null,
    'View',
    '/pages/listings/listing/index.html?id=' + id ,
  );`

**Comment by Tokle89** [comment]
have create a setTimeout function for when an id is not included. then the user is shown an aler message, and send back to the listings page

File: src/js/templates/listings/renderListing.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
+import { getSingleListing } from '../../api/posts/getSingleListing.js';
+
+import { createElement } from '../CreateHtml.js';
+import { parseDate } from '../../utilities/parse/parse.js';
+import { findDaysAgo } from '../../utilities/dateConverter/dateConverter.js';
+import { set } from 'immutable';
```
Comment (Olekrr) [inline src/js/templates/listings/renderListing.js:6]: What is this?

**Review by Olekrr — APPROVED** [review]
After a long run this is finally ready for approval and merging :)

**Pull request merged**